ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs

```diff
@@ -521,3 +543,47 @@ translateValidatedTxAllegraToMaryWrapper ::
        (ShelleyBlock (MaryEra c))
 translateValidatedTxAllegraToMaryWrapper = InjectValidatedTx $
     fmap unComp . eitherToMaybe . runExcept . SL.translateEra () . Comp
+
+{-------------------------------------------------------------------------------
+  Translation from Mary to Alonzo
+-------------------------------------------------------------------------------}
+
+translateLedgerStateMaryToAlonzoWrapper ::
+     PraosCrypto c
+  => RequiringBoth
+       WrapLedgerConfig
+       (Translate LedgerState)
+       (ShelleyBlock (MaryEra c))
+       (ShelleyBlock (AlonzoEra c))
+translateLedgerStateMaryToAlonzoWrapper =
+    RequireBoth $ \_cfgMary cfgAlonzo ->
+      Translate $ \_epochNo ->
+        unComp . SL.translateEra' (getAlonzoTranslationContext cfgAlonzo) . Comp
+
+getAlonzoTranslationContext ::
+     WrapLedgerConfig (ShelleyBlock (AlonzoEra c))
+  -> Alonzo.AlonzoGenesis
+getAlonzoTranslationContext =
+    shelleyLedgerTranslationContext . unwrapLedgerConfig
+
+translateTxMaryToAlonzoWrapper ::
+     PraosCrypto c
+  => Alonzo.AlonzoGenesis
```

Comment:
I'm confused by the fact that `getAlonzoTranslationContext` returns sth that we name as `ctxt` but it is specifically an `Alonzo.AlonzoGenesis`. From what I can see, `shelleyLedgerTranslationContext :: !(Core.TranslationContext era)`, which is associated on their side as:
```
type instance TranslationContext (AlonzoEra c) = AlonzoGenesis
```

I think it would be clearer to make this receive an argument of type `TranslationContext (AlonzoEra c)` and in an abstract way give it to `SL.translateEra` as we "don't care" what that type is from our side. We just use it to give it to this function, and it is a `TranslationContext` and we name it `ctxt`.

Reply:
I wrote `AlonzoGenesis` instead of `TranslationContext (AlonzoEra c)` because I _generally_ avoid writing type family applications that can be immediately simplified. (And not just avoiding beta reduxes; I mean the result is _simpler_.)

But I did name the function and the variables "context", since that is _why_ it's being passed.

I admit that's a mismash of choices I've made in our code here.

Thanks for the note, I'll consider it with Pawel later day.

Reply:
Pawel and I decided to leave it as is. That's both of our intuitions. We discussed a few reasons, which we'll talk to you about. But mainly: there's a slight modularity gain to doing what you said, but at least in this specific instance, we're favoring the transparency of using the Alonzo-specific types in these Alonzo-specific contexts.